ocp_resources/datavolume.py

```diff
@@ -308,7 +308,7 @@ def test_dv():
                 wait_timeout=timeout,
                 func=lambda: self.exists,
             ):
-                if dv_garbage_collection_enabled is not None:
+                if dv_garbage_collection_enabled:
```

Comment:
why limit? what is someone sends `False`?

Reply:
Do not remove `is not None` please.
Add another condition inside to show deprecated warn only if `dv_garbage_collection_enabled:`

Reply:
But, if someone pass explicit `False` he will not get this warn so I think that how it's now (without your change) is OK

Reply:
Right now because `dv_garbage_collection_enabled=False` by default, all wait_for_dv_success() calls are showing this warning even though they are not explicitly using this argument. It is too many warnings for no reason. What do you suggest the caller do?

Reply:
Anyone using this class and passing `dv_garbage_collection_enabled` (or depends on the default value) should be warned